docs/02-app/01-building-your-application/08-testing/02-jest.mdx

```diff
@@ -0,0 +1,387 @@
+---
+title: Setting up Jest with Next.js
+nav: Jest
+description: Learn how to set up Jest with Next.js for Unit Testing and Snapshot Testing.
+---
+
+Jest and React Testing Library are frequently used together for **Unit Testing** and **Snapshot Testing**. This guide will show you how to set up Jest with Next.js and write your first tests.
+
+> **Good to know:** Since `async` Server Components are new to the React ecosystem, Jest currently does not support them. While you can still run **unit tests** for synchronous Server and Client Components, we recommend using an **E2E tests** for `async` components.
+
+## Quickstart
+
+You can use `create-next-app` with the Next.js [with-jest](https://github.com/vercel/next.js/tree/canary/examples/with-jest) example to quickly get started:
+
+```bash filename="Terminal"
+npx create-next-app@latest --example with-jest with-jest-app
```

Comment:
The Vitest one had npm/yarn/pnpm, but no bun, should we just pick one or list them all?

Reply:
Let's see how they look like on the site, we could add a switcher.